data/replication data/DICE2016R2-Unc-base-v5-55555-083017B.gms

```diff
@@ -0,0 +1,506 @@
+        a20 = a2base;
+        sig0 = e0/(q0*(1-miu0));
+
+        l("1") = pop0;
+        loop(t, l(t+1)=l(t););
+        loop(t, l(t+1)=l(t)*(popasym/L(t))**popadj ;);
+
+        gsig("1")=gsigma1; loop(t,gsig(t+1)=gsig(t)*((1+dsig)**tstep) ;);
+        sigma("1")=sig0;   loop(t,sigma(t+1)=(sigma(t)*exp(gsig(t)*tstep)););
+
+        pbacktime(t)=pback*(1-gback)**(t.val-1);
+        cost1(t) = pbacktime(t)*sigma(t)/expcost2/1000;
+
+        etree(t) = eland0*(1-deland)**(t.val-1);
+        cumetree("1")= 100; loop(t,cumetree(t+1)=cumetree(t)+etree(t)*(5/3.666););
+        rr(t) = 1/((1+prstp)**(tstep*(t.val-1)));
+        forcoth(t) = fex0+ (1/17)*(fex1-fex0)*(t.val-1)$(t.val lt 18)+ (fex1-fex0)$(t.val ge 18);
+        optlrsav = (dk + .004)/(dk + .004*elasmu + prstp)*gama;
+
+
+*Base Case      Carbon Price
+        cpricebase(t)= cprice0*(1+gcprice)**(5*(t.val-1));
+
+VARIABLES
+        MIU(t)          Emission control rate GHGs
+        FORC(t)         Increase in radiative forcing (watts per m2 from 1900)
+        TATM(t)         Increase temperature of atmosphere (degrees C from 1900)
+        TOCEAN(t)       Increase temperatureof lower oceans (degrees C from 1900)
+        MAT(t)          Carbon concentration increase in atmosphere (GtC from 1750)
+        MU(t)           Carbon concentration increase in shallow oceans (GtC from 1750)
+        ML(t)           Carbon concentration increase in lower oceans (GtC from 1750)
+        E(t)            Total CO2 emissions (GtCO2 per year)
+        EIND(t)         Industrial emissions (GtCO2 per year)
+        C(t)            Consumption (trillions 2005 US dollars per year)
+        K(t)            Capital stock (trillions 2005 US dollars)
+        CPC(t)          Per capita consumption (thousands 2005 USD per year)
+        I(t)            Investment (trillions 2005 USD per year)
+        S(t)            Gross savings rate as fraction of gross world product
+        RI(t)           Real interest rate (per annum)
+        Y(t)            Gross world product net of abatement and damages (trillions 2005 USD per year)
+        YGROSS(t)       Gross world product GROSS of abatement and damages (trillions 2005 USD per year)
+        YNET(t)         Output net of damages equation (trillions 2005 USD per year)
+        DAMAGES(t)      Damages (trillions 2005 USD per year)
+        DAMFRAC(t)      Damages as fraction of gross output
+        ABATECOST(t)    Cost of emissions reductions  (trillions 2005 USD per year)
+        MCABATE(t)      Marginal cost of abatement (2005$ per ton CO2)
+        CCA(t)          Cumulative industrial carbon emissions (GTC)
+        CCATOT(t)       Total carbon emissions (GtC)
+        PERIODU(t)      One period utility function
+
+        CPRICE(t)       Carbon price (2005$ per ton of CO2)
+        CEMUTOTPER(t)   Period utility
+        UTILITY         Welfare function
+;
+
+POSITIVE VARIABLES  MIU, TATM, MAT, MU, ML, Y, YGROSS, C, K, I;
```

Comment:
@davidanthoff actually, I just found this line in the GAS file. I guess this is confirmation that MAT can't go negative? and Y and YGROSS and C, for that matter :)

Reply:
maybe this explicit bound should be added in the code everywhere?